Space stops back-search with zsh

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### What happened?

I've been trying out omp with zsh and there is one weird issue: when I trigger back-search (reverse search) with Ctrl-R and type some words, the search is "stopped" when I press the space bar.

### Theme

100% default configuration.

### What OS are you seeing the problem on?

Linux

### Which shell are you using?

zsh

### Log output

```Shell
Version: 21.5.0

Shell: zsh

Prompt:

 felix   ~    

Segments:

ConsoleTitle(true)                        -  60 ms
Session(true)                             -   6 ms
Path(true)                                -   6 ms
Git(false)                                -   2 ms
Root(false)                               -   0 ms
Status(true)                              -   7 ms
Node(false)                               -   0 ms
Go(false)                                 -   0 ms
Python(false)                             -   0 ms
Shell(true)                               -   0 ms
Time(true)                                -   0 ms

Run duration: 77.924155ms

Cache path: /home/felix/.cache/oh-my-posh

Config path: no --config set, using default built-in configuration

Logs:

[DEBUG] 14:07:22.651 shell.go:Getenv:326 → NO DATA
[TRACE] 14:07:22.651 shell.go:Getenv(TERM_PROGRAM) - 40.679µs
[DEBUG] 14:07:22.651 shell.go:Getenv:326 → NO DATA
[TRACE] 14:07:22.651 shell.go:Getenv(OMP_CACHE_DIR) - 2.444µs
[DEBUG] 14:07:22.651 shell.go:Getenv:326 → NO DATA
[TRACE] 14:07:22.651 shell.go:Getenv(XDG_CACHE_HOME) - 1.814µs
[TRACE] 14:07:22.651 shell_unix.go:CachePath() - 19.867µs
[DEBUG] 14:07:22.653 shell.go:Getenv:326 → NO DATA
[TRACE] 14:07:22.653 shell.go:Getenv(POSH_THEME) - 6.038µs
[DEBUG] 14:07:22.653 shell.go:resolveConfigPath:246 → No config set, fallback to default config
[TRACE] 14:07:22.653 shell.go:resolveConfigPath() - 53.202µs
[TRACE] 14:07:22.653 shell.go:Init() - 1.416656ms
[TRACE] 14:07:22.653 shell.go:Flags() - 38ns
[DEBUG] 14:07:22.653 config.go:loadConfig:128 → no config file specified, using default
[TRACE] 14:07:22.653 config.go:loadConfig() - 9.714µs
[TRACE] 14:07:22.653 shell.go:Flags() - 26ns
[DEBUG] 14:07:22.653 shell.go:Getenv:326 → NO DATA
[TRACE] 14:07:22.653 shell.go:Getenv(OMP_CACHE_DISABLED) - 2.182µs
[DEBUG] 14:07:22.653 shell.go:Shell:601 → no shell name provided in flags, trying to detect it
[DEBUG] 14:07:22.653 shell.go:Shell:609 → process name: zsh
[TRACE] 14:07:22.653 shell.go:Shell() - 655.527µs
[DEBUG] 14:07:22.653 shell.go:Getenv:326 → NO DATA
[TRACE] 14:07:22.653 shell.go:Getenv(POSH_SHELL_VERSION) - 4.011µs
[DEBUG] 14:07:22.653 debug.go:PrintDebug:22 → Segment: Title
[DEBUG] 14:07:22.712 text.go:Render:71 → Rendering template: {{ .Shell }} in {{ .Folder }}
[TRACE] 14:07:22.713 shell_unix.go:Root() - 3.794µs
[TRACE] 14:07:22.713 shell.go:Shell() - 433ns
[TRACE] 14:07:22.713 shell.go:StatusCodes() - 331ns
[DEBUG] 14:07:22.713 shell_unix.go:IsWsl:35 → false
[TRACE] 14:07:22.713 shell_unix.go:IsWsl() - 44.369µs
[DEBUG] 14:07:22.714 shell.go:TemplateCache:782 → environment: [PATH=/usr/lib/ccache/bin:/home/felix/bin:/home/felix/.local/bin:/home/felix/bin/git_commands:/home/felix/.cargo/bin:/home/felix/.ruby/bin:/home/felix/.local/share/gem/ruby/3.0.0/bin:/home/felix/bin:/usr/local/bin:/usr/local/sbin:/home/felix/.local/bin:/usr/bin:/bin:/usr/lib/jvm/default/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl:/usr/lib/rustup/bin:/var/lib/snapd/snap/bin PAGER=less GDMSESSION=cinnamon LANG=fr_FR.UTF-8 DEBUGINFOD_URLS=https://debuginfod.archlinux.org  GTK3_MODULES=xapp-gtk3-module XDG_CURRENT_DESKTOP=X-Cinnamon SAL_USE_VCLPLUGIN=gtk LC_IDENTIFICATION=fr_FR.UTF-8 GIO_LAUNCHED_DESKTOP_FILE_PID=1275 XDG_GREETER_DATA_DIR=/var/lib/lightdm-data/felix SESSION_MANAGER=local/samson:@/tmp/.ICE-unix/1236,unix/samson:/tmp/.ICE-unix/1236 USER=felix LC_MEASUREMENT=fr_FR.UTF-8 OLDPWD=/home/felix XDG_SESSION_PATH=/org/freedesktop/DisplayManager/Session0 GNOME_KEYRING_CONTROL=/run/user/1000/keyring DESKTOP_SESSION=cinnamon TERM=xterm-256color HOME=/home/felix XDG_VTNR=7 XDG_SEAT=seat0 DBUS_SESSION_BUS_ADDRESS=unix:path=/run/user/1000/bus LESS=-g -i -M -R -S -w -X -z-4 GTK_MODULES=canberra-gtk-module XDG_DATA_DIRS=/home/felix/.local/share/flatpak/exports/share:/var/lib/flatpak/exports/share:/usr/local/share:/usr/share:/var/lib/snapd/desktop VTE_VERSION=7602 XDG_SESSION_DESKTOP=cinnamon LC_NUMERIC=fr_FR.UTF-8 GNOME_DESKTOP_SESSION_ID=this-is-deprecated LC_TIME=fr_FR.UTF-8 MAIL=/var/spool/mail/felix XDG_SEAT_PATH=/org/freedesktop/DisplayManager/Seat0 LOGNAME=felix VISUAL=nano LC_PAPER=fr_FR.UTF-8 XDG_RUNTIME_DIR=/run/user/1000 TILIX_ID=a4a21157-881b-4272-96fb-48d2d66cee20 SHELL=/usr/bin/zsh XDG_SESSION_TYPE=x11 LC_MONETARY=fr_FR.UTF-8 XDG_SESSION_ID=2 LC_TELEPHONE=fr_FR.UTF-8 EDITOR=xed GIO_LAUNCHED_DESKTOP_FILE=/etc/xdg/autostart/cinnamon-settings-daemon-media-keys.desktop MOTD_SHOWN=pam PWD=/home/felix XAUTHORITY=/home/felix/.Xauthority COLORTERM=truecolor LC_NAME=fr_FR.UTF-8 XDG_SESSION_CLASS=user DISPLAY=:0 SHLVL=1 LC_ADDRESS=fr_FR.UTF-8 LESS_TERMCAP_mb= LESS_TERMCAP_md= LESS_TERMCAP_me= LESS_TERMCAP_se= LESS_TERMCAP_so= LESS_TERMCAP_ue= LESS_TERMCAP_us= LS_COLORS=rs=0:di=01;34:ln=01;36:mh=00:pi=40;33:so=01;35:do=01;35:bd=40;33;01:cd=40;33;01:or=01;05;37;41:mi=01;05;37;41:su=37;41:sg=30;43:ca=30;41:tw=30;42:ow=34;42:st=37;44:ex=01;32:*.tar=01;31:*.tgz=01;31:*.arc=01;31:*.arj=01;31:*.taz=01;31:*.lha=01;31:*.lz4=01;31:*.lzh=01;31:*.lzma=01;31:*.tlz=01;31:*.txz=01;31:*.tzo=01;31:*.t7z=01;31:*.zip=01;31:*.z=01;31:*.Z=01;31:*.dz=01;31:*.gz=01;31:*.lrz=01;31:*.lz=01;31:*.lzo=01;31:*.xz=01;31:*.bz2=01;31:*.bz=01;31:*.tbz=01;31:*.tbz2=01;31:*.tz=01;31:*.deb=01;31:*.rpm=01;31:*.jar=01;31:*.war=01;31:*.ear=01;31:*.sar=01;31:*.rar=01;31:*.alz=01;31:*.ace=01;31:*.zoo=01;31:*.cpio=01;31:*.7z=01;31:*.rz=01;31:*.cab=01;31:*.jpg=01;35:*.jpeg=01;35:*.gif=01;35:*.bmp=01;35:*.pbm=01;35:*.pgm=01;35:*.ppm=01;35:*.tga=01;35:*.xbm=01;35:*.xpm=01;35:*.tif=01;35:*.tiff=01;35:*.png=01;35:*.svg=01;35:*.svgz=01;35:*.mng=01;35:*.pcx=01;35:*.mov=01;35:*.mpg=01;35:*.mpeg=01;35:*.m2v=01;35:*.mkv=01;35:*.webm=01;35:*.ogm=01;35:*.mp4=01;35:*.m4v=01;35:*.mp4v=01;35:*.vob=01;35:*.qt=01;35:*.nuv=01;35:*.wmv=01;35:*.asf=01;35:*.rm=01;35:*.rmvb=01;35:*.flc=01;35:*.avi=01;35:*.fli=01;35:*.flv=01;35:*.gl=01;35:*.dl=01;35:*.xcf=01;35:*.xwd=01;35:*.yuv=01;35:*.cgm=01;35:*.emf=01;35:*.axv=01;35:*.anx=01;35:*.ogv=01;35:*.ogx=01;35:*.pdf=00;32:*.ps=00;32:*.txt=00;32:*.patch=00;32:*.diff=00;32:*.log=00;32:*.tex=00;32:*.doc=00;32:*.aac=00;36:*.au=00;36:*.flac=00;36:*.m4a=00;36:*.mid=00;36:*.midi=00;36:*.mka=00;36:*.mp3=00;36:*.mpc=00;36:*.ogg=00;36:*.ra=00;36:*.wav=00;36:*.axa=00;36:*.oga=00;36:*.spx=00;36:*.xspf=00;36: GREP_COLOR=37;45 GREP_COLORS=mt=37;45 BUNDLE_PATH=/home/felix/.gems GEM_HOME=/home/felix/.ruby NVM_DIR=/home/felix/.nvm QT_QPA_PLATFORMTHEME=qt5ct GTK2_RC_FILES=/home/felix/.gtkrc-2.0 TMUX_XPANES_TMUX_VERSION=3.2 _=/usr/bin/oh-my-posh]
[TRACE] 14:07:22.714 shell.go:GOOS() - 250ns
[DEBUG] 14:07:22.714 shell.go:Pwd:356 → /home/felix
[TRACE] 14:07:22.714 shell.go:Pwd() - 58.327µs
[TRACE] 14:07:22.714 shell.go:GOOS() - 105ns
[TRACE] 14:07:22.714 shell.go:GOOS() - 100ns
[TRACE] 14:07:22.714 shell.go:GOOS() - 100ns
[TRACE] 14:07:22.714 shell.go:GOOS() - 126ns
[TRACE] 14:07:22.714 shell.go:GOOS() - 127ns
[TRACE] 14:07:22.714 shell.go:GOOS() - 103ns
[TRACE] 14:07:22.714 shell.go:GOOS() - 100ns
[DEBUG] 14:07:22.714 shell.go:User:494 → felix
[TRACE] 14:07:22.714 shell.go:User() - 10.513µs
[DEBUG] 14:07:22.714 shell.go:Host:511 → samson
[TRACE] 14:07:22.714 shell.go:Host() - 14.866µs
[TRACE] 14:07:22.714 shell.go:GOOS() - 96ns
[DEBUG] 14:07:22.714 shell_unix.go:Platform:87 → manjaro
[DEBUG] 14:07:22.714 shell.go:Getenv:326 → 1
[TRACE] 14:07:22.714 shell.go:Getenv(SHLVL) - 9.524µs
[TRACE] 14:07:22.714 shell.go:TemplateCache() - 577.25µs
[DEBUG] 14:07:22.714 shell.go:Getenv:326 → NO DATA
[TRACE] 14:07:22.714 shell.go:Getenv(POSH_CURSOR_LINE) - 9.476µs
[DEBUG] 14:07:22.714 shell.go:Getenv:326 → NO DATA
[TRACE] 14:07:22.714 shell.go:Getenv(POSH_CURSOR_COLUMN) - 7.919µs
[TRACE] 14:07:22.714 shell.go:Flags() - 191ns
[TRACE] 14:07:22.714 shell.go:Flags() - 103ns
[TRACE] 14:07:22.714 shell.go:Shell() - 160ns
[TRACE] 14:07:22.714 shell.go:Shell() - 101ns
[TRACE] 14:07:22.714 shell.go:Flags() - 125ns
[TRACE] 14:07:22.714 shell.go:Flags() - 629ns
[TRACE] 14:07:22.714 shell.go:Pwd() - 445ns
[TRACE] 14:07:22.714 shell.go:GOOS() - 206ns
[DEBUG] 14:07:22.714 segment.go:SetEnabled:536 → Segment: Path
[TRACE] 14:07:22.714 shell.go:Pwd() - 159ns
[TRACE] 14:07:22.714 shell.go:Shell() - 191ns
[TRACE] 14:07:22.714 shell.go:Shell() - 112ns
[TRACE] 14:07:22.715 shell.go:Flags() - 527ns
[TRACE] 14:07:22.715 shell.go:Pwd() - 474ns
[TRACE] 14:07:22.715 shell.go:GOOS() - 150ns
[DEBUG] 14:07:22.715 segment.go:SetEnabled:536 → Segment: Git
[TRACE] 14:07:22.715 shell.go:Flags() - 634ns
[DEBUG] 14:07:22.715 shell_unix.go:IsWsl:35 → false
[TRACE] 14:07:22.715 shell_unix.go:IsWsl() - 12.419µs
[TRACE] 14:07:22.715 shell_unix.go:IsWsl2() - 22.916µs
[TRACE] 14:07:22.715 shell.go:GOOS() - 180ns
[TRACE] 14:07:22.715 shell.go:Pwd() - 476ns
[TRACE] 14:07:22.715 shell.go:GOOS() - 219ns
[DEBUG] 14:07:22.715 properties.go:GetString:28 → {{ .Code }}
[TRACE] 14:07:22.715 shell.go:Pwd() - 764ns
[DEBUG] 14:07:22.717 shell.go:Getenv:326 → NO DATA
[TRACE] 14:07:22.717 shell.go:Pwd() - 477ns
[TRACE] 14:07:22.717 shell.go:Getenv(SSH_CLIENT) - 11.669µs
[DEBUG] 14:07:22.717 shell_unix.go:Platform:87 → manjaro
[TRACE] 14:07:22.717 shell.go:Flags() - 605ns
[ERROR] 14:07:22.717 shell.go:HasParentFilePath:697 → stat .git: no such file or directory
[TRACE] 14:07:22.717 shell.go:HasParentFilePath(.git) - 122.765µs
[TRACE] 14:07:22.717 shell.go:Pwd() - 814ns
[DEBUG] 14:07:22.717 properties.go:GetBool:22 → fetch_bare_info: false
[TRACE] 14:07:22.717 shell.go:GOOS() - 225ns
[TRACE] 14:07:22.717 shell.go:GOOS() - 507ns
[DEBUG] 14:07:22.717 segment.go:SetEnabled:536 → Segment: Root
[DEBUG] 14:07:22.715 properties.go:GetBool:22 → mapped_locations_enabled: true
[DEBUG] 14:07:22.717 segment.go:SetEnabled:536 → Segment: Status
[TRACE] 14:07:22.717 shell.go:StatusCodes() - 502ns
[TRACE] 14:07:22.717 shell_unix.go:Root() - 3.49µs
[DEBUG] 14:07:22.717 text.go:Render:71 → Rendering template: {{ .Code }}
[DEBUG] 14:07:22.717 properties.go:GetString:28 → 
[DEBUG] 14:07:22.717 properties.go:GetString:28 → 
[TRACE] 14:07:22.717 shell.go:GOOS() - 446ns
[DEBUG] 14:07:22.717 properties.go:GetString:28 → ~
[DEBUG] 14:07:22.717 properties.go:GetKeyValueMap:46 → mapped_locations: map[]
[TRACE] 14:07:22.717 shell.go:PathSeparator() - 274ns
[TRACE] 14:07:22.717 shell.go:GOOS() - 195ns
[TRACE] 14:07:22.717 shell.go:GOOS() - 185ns
[TRACE] 14:07:22.721 shell.go:GOOS() - 592ns
[TRACE] 14:07:22.721 shell.go:GOOS() - 101ns
[TRACE] 14:07:22.721 shell.go:PathSeparator() - 166ns
[TRACE] 14:07:22.721 shell.go:PathSeparator() - 99ns
[TRACE] 14:07:22.721 shell.go:GOOS() - 100ns
[TRACE] 14:07:22.721 shell.go:GOOS() - 146ns
[TRACE] 14:07:22.721 shell.go:PathSeparator() - 165ns
[TRACE] 14:07:22.721 shell.go:GOOS() - 108ns
[TRACE] 14:07:22.721 shell.go:GOOS() - 101ns
[TRACE] 14:07:22.721 shell.go:PathSeparator() - 100ns
[TRACE] 14:07:22.721 shell.go:GOOS() - 102ns
[TRACE] 14:07:22.721 shell.go:GOOS() - 101ns
[TRACE] 14:07:22.721 shell.go:PathSeparator() - 100ns
[DEBUG] 14:07:22.721 shell.go:RunCommand:531 → NO DATA
[DEBUG] 14:07:22.721 properties.go:GetString:28 → NO DATA
[TRACE] 14:07:22.721 shell.go:PathSeparator() - 99ns
[TRACE] 14:07:22.721 shell.go:Pwd() - 405ns
[TRACE] 14:07:22.721 shell.go:RunCommand(who am i) - 3.898038ms
[DEBUG] 14:07:22.721 shell_unix.go:IsWsl:35 → false
[TRACE] 14:07:22.721 shell_unix.go:IsWsl() - 14.309µs
[TRACE] 14:07:22.721 shell.go:StackCount() - 301ns
[TRACE] 14:07:22.721 shell_unix.go:DirIsWritable(/home/felix) - 19.509µs
[TRACE] 14:07:22.721 shell.go:TemplateCache() - 632ns
[TRACE] 14:07:22.721 shell.go:TemplateCache() - 704ns
[TRACE] 14:07:22.721 shell.go:TemplateCache() - 1.061µs
[DEBUG] 14:07:22.721 properties.go:GetBool:22 → always_enabled: true
[TRACE] 14:07:22.721 shell.go:TemplateCache() - 235ns
[DEBUG] 14:07:22.721 text.go:Render:71 → Rendering template:  {{ if gt .Code 0 }}{{ else }}{{ end }} 
[DEBUG] 14:07:22.721 text.go:Render:71 → Rendering template:   {{ path .Path .Location }} 
[DEBUG] 14:07:22.721 text.go:Render:71 → Rendering template:  {{ if .SSHSession }} {{ end }}{{ .UserName }} 
[TRACE] 14:07:22.722 shell.go:TemplateCache() - 443ns
[TRACE] 14:07:22.722 shell.go:Shell() - 259ns
[TRACE] 14:07:22.722 shell.go:TemplateCache() - 1.08µs
[TRACE] 14:07:22.723 shell.go:Shell() - 491ns
[TRACE] 14:07:22.723 shell.go:TemplateCache() - 703ns
[TRACE] 14:07:22.723 shell.go:Shell() - 250ns
[DEBUG] 14:07:22.723 text.go:Render:71 → Rendering template: diamond
[DEBUG] 14:07:22.723 text.go:Render:71 → Rendering template: powerline
[DEBUG] 14:07:22.723 text.go:Render:71 → Rendering template: powerline
[DEBUG] 14:07:22.723 text.go:Render:71 → Rendering template: powerline
[DEBUG] 14:07:22.723 text.go:Render:71 → Rendering template: {{ if gt .Code 0 }}p:red{{ end }}
[TRACE] 14:07:22.726 shell.go:TemplateCache() - 1.175µs
[DEBUG] 14:07:22.727 text.go:Render:71 → Rendering template: diamond
[TRACE] 14:07:22.727 shell.go:Shell() - 767ns
[TRACE] 14:07:22.727 shell.go:Shell() - 254ns
[TRACE] 14:07:22.727 shell.go:Shell() - 136ns
[TRACE] 14:07:22.727 shell.go:Shell() - 254ns
[TRACE] 14:07:22.727 shell.go:Flags() - 353ns
[TRACE] 14:07:22.727 shell.go:Pwd() - 539ns
[TRACE] 14:07:22.727 shell.go:GOOS() - 292ns
[DEBUG] 14:07:22.727 segment.go:SetEnabled:536 → Segment: Node
[TRACE] 14:07:22.727 shell.go:Flags() - 608ns
[TRACE] 14:07:22.727 shell.go:Flags() - 558ns
[DEBUG] 14:07:22.727 properties.go:GetStringArray:52 → extensions: [*.js *.ts package.json .nvmrc pnpm-workspace.yaml .pnpmfile.cjs .vue]
[DEBUG] 14:07:22.727 properties.go:GetString:28 → files
[TRACE] 14:07:22.727 shell.go:Pwd() - 465ns
[TRACE] 14:07:22.727 shell.go:GOOS() - 208ns
[DEBUG] 14:07:22.727 properties.go:GetStringArray:52 → folders: []
[TRACE] 14:07:22.727 shell.go:Pwd() - 550ns
[DEBUG] 14:07:22.727 segment.go:SetEnabled:536 → Segment: Shell
[TRACE] 14:07:22.727 shell.go:GOOS() - 210ns
[DEBUG] 14:07:22.727 properties.go:GetBool:22 → home_enabled: false
[DEBUG] 14:07:22.727 properties.go:GetKeyValueMap:46 → mapped_shell_names: map[]
[DEBUG] 14:07:22.727 segment.go:SetEnabled:536 → Segment: Python
[TRACE] 14:07:22.727 shell.go:Pwd() - 402ns
[TRACE] 14:07:22.727 shell.go:Shell() - 200ns
[TRACE] 14:07:22.727 shell.go:Flags() - 174ns
[TRACE] 14:07:22.727 shell.go:TemplateCache() - 623ns
[TRACE] 14:07:22.727 shell.go:Flags() - 562ns
[DEBUG] 14:07:22.727 properties.go:GetStringArray:52 → extensions: [*.py *.ipynb pyproject.toml venv.bak]
[DEBUG] 14:07:22.727 properties.go:GetStringArray:52 → folders: [.venv venv virtualenv venv-win pyenv-win]
[TRACE] 14:07:22.727 shell.go:Pwd() - 494ns
[DEBUG] 14:07:22.727 properties.go:GetBool:22 → home_enabled: false
[TRACE] 14:07:22.727 shell.go:GOOS() - 212ns
[TRACE] 14:07:22.727 shell.go:Flags() - 592ns
[TRACE] 14:07:22.727 shell.go:Pwd() - 202ns
[DEBUG] 14:07:22.727 segment.go:SetEnabled:536 → Segment: Go
[DEBUG] 14:07:22.727 properties.go:GetStringArray:52 → extensions: [*.go go.mod]
[TRACE] 14:07:22.727 shell.go:Pwd() - 517ns
[TRACE] 14:07:22.727 shell.go:Pwd() - 237ns
[TRACE] 14:07:22.727 shell.go:GOOS() - 248ns
[DEBUG] 14:07:22.728 segment.go:SetEnabled:536 → Segment: Time
[DEBUG] 14:07:22.728 properties.go:GetString:28 → 15:04:05
[TRACE] 14:07:22.728 shell.go:TemplateCache() - 638ns
[DEBUG] 14:07:22.728 text.go:Render:71 → Rendering template: at <p:blue><b>{{ .CurrentDate | date "15:04:05" }}</b></>
[DEBUG] 14:07:22.728 text.go:Render:71 → Rendering template: in <p:blue><b>{{ .Name }}</b></> 
[TRACE] 14:07:22.728 shell.go:TemplateCache() - 684ns
[TRACE] 14:07:22.728 shell.go:Shell() - 307ns
[TRACE] 14:07:22.728 shell.go:TemplateCache() - 1.043µs
[TRACE] 14:07:22.728 shell.go:Shell() - 287ns
[DEBUG] 14:07:22.728 text.go:Render:71 → Rendering template: plain
[DEBUG] 14:07:22.728 text.go:Render:71 → Rendering template: plain
[DEBUG] 14:07:22.729 text.go:Render:71 → Rendering template: plain
[DEBUG] 14:07:22.729 text.go:Render:71 → Rendering template: plain
[DEBUG] 14:07:22.729 text.go:Render:71 → Rendering template: plain
[TRACE] 14:07:22.729 shell.go:Shell() - 644ns
[TRACE] 14:07:22.729 shell.go:Shell() - 233ns
[TRACE] 14:07:22.729 shell.go:Flags() - 208ns
[TRACE] 14:07:22.729 shell.go:Shell() - 156ns
[TRACE] 14:07:22.729 shell.go:Shell() - 246ns
[TRACE] 14:07:22.729 shell.go:Flags() - 112ns
[DEBUG] 14:07:22.729 shell.go:Getenv:326 → NO DATA
[TRACE] 14:07:22.729 shell.go:Getenv(OMP_CACHE_DIR) - 12.799µs
[DEBUG] 14:07:22.729 shell.go:Getenv:326 → NO DATA
[TRACE] 14:07:22.729 shell.go:Getenv(XDG_CACHE_HOME) - 8.354µs
[TRACE] 14:07:22.729 shell_unix.go:CachePath() - 82.827µs
[TRACE] 14:07:22.729 shell.go:Flags() - 131ns
```

--- a/src/shell/scripts/omp.bash
+++ b/src/shell/scripts/omp.bash
@@ -4,19 +4,19 @@ export POWERLINE_COMMAND="oh-my-posh"
 export POSH_PID=$$
 export CONDA_PROMPT_MODIFIER=false
 
-omp_start_time=""
-omp_stack_count=0
-omp_elapsed=-1
-omp_no_exit_code="true"
-omp_ret=0
-omp_pipe_status=0
+_omp_start_time=""
+_omp_stack_count=0
+_omp_elapsed=-1
+_omp_no_exit_code="true"
+_omp_status_cache=0
+_omp_pipestatus_cache=0
 
 # start timer on command start
-PS0='${omp_start_time:0:$((omp_start_time="$(_omp_start_timer)",0))}$(_omp_ftcs_command_start)'
+PS0='${_omp_start_time:0:$((_omp_start_time="$(_omp_start_timer)",0))}$(_omp_ftcs_command_start)'
 # set secondary prompt
 PS2="$(::OMP:: print secondary --config="$POSH_THEME" --shell=bash --shell-version="$BASH_VERSION")"
 
-function _set_posh_cursor_position() {
+function _omp_set_cursor_position() {
     # not supported in Midnight Commander
     # see https://github.com/JanDeDobbeleer/oh-my-posh/issues/3415
     if [[ "::CURSOR::" != "true" ]] || [[ -v MC_SID ]]; then
@@ -53,35 +53,35 @@ function set_poshcontext() {
 
 # regular prompt
 function _omp_hook() {
-    omp_ret=$? omp_pipe_status=(${PIPESTATUS[@]})
+    _omp_status_cache=$? _omp_pipestatus_cache=(${PIPESTATUS[@]})
 
-    if [[ "${#BP_PIPESTATUS[@]}" -ge "${#omp_pipe_status[@]}" ]]; then
-        omp_pipe_status=(${BP_PIPESTATUS[@]})
+    if [[ "${#BP_PIPESTATUS[@]}" -ge "${#_omp_pipestatus_cache[@]}" ]]; then
+        _omp_pipestatus_cache=(${BP_PIPESTATUS[@]})
     fi
 
-    omp_stack_count=$((${#DIRSTACK[@]} - 1))
+    _omp_stack_count=$((${#DIRSTACK[@]} - 1))
 
-    if [[ "$omp_start_time" ]]; then
+    if [[ "$_omp_start_time" ]]; then
         local omp_now=$(::OMP:: get millis --shell=bash)
-        omp_elapsed=$((omp_now - omp_start_time))
-        omp_start_time=""
-        omp_no_exit_code="false"
+        _omp_elapsed=$((omp_now - _omp_start_time))
+        _omp_start_time=""
+        _omp_no_exit_code="false"
     fi
 
-    if [[ "${omp_pipe_status[-1]}" != "$omp_ret" ]]; then
-        omp_pipe_status=("$omp_ret")
+    if [[ "${_omp_pipestatus_cache[-1]}" != "$_omp_status_cache" ]]; then
+        _omp_pipestatus_cache=("$_omp_status_cache")
     fi
 
     set_poshcontext
-    _set_posh_cursor_position
+    _omp_set_cursor_position
 
-    PS1="$(::OMP:: print primary --config="$POSH_THEME" --shell=bash --shell-version="$BASH_VERSION" --status="$omp_ret" --pipestatus="${omp_pipe_status[*]}" --execution-time="$omp_elapsed" --stack-count="$omp_stack_count" --no-status="$omp_no_exit_code" --terminal-width="${COLUMNS-0}" | tr -d '\0')"
+    PS1="$(::OMP:: print primary --config="$POSH_THEME" --shell=bash --shell-version="$BASH_VERSION" --status="$_omp_status_cache" --pipestatus="${_omp_pipestatus_cache[*]}" --execution-time="$_omp_elapsed" --stack-count="$_omp_stack_count" --no-status="$_omp_no_exit_code" --terminal-width="${COLUMNS-0}" | tr -d '\0')"
 }
 
 # rprompt
-_omp_rprompt() {
-    ::OMP:: print right --config="$POSH_THEME" --shell=bash --shell-version="$BASH_VERSION" --status="$omp_ret" --pipestatus="${omp_pipe_status[*]}" --execution-time="$omp_elapsed" --stack-count="$omp_stack_count" --no-status="$omp_no_exit_code" --terminal-width="${COLUMNS-0}" | tr -d '\0'
-    return $omp_ret
+function _omp_rprompt() {
+    ::OMP:: print right --config="$POSH_THEME" --shell=bash --shell-version="$BASH_VERSION" --status="$_omp_status_cache" --pipestatus="${_omp_pipestatus_cache[*]}" --execution-time="$_omp_elapsed" --stack-count="$_omp_stack_count" --no-status="$_omp_no_exit_code" --terminal-width="${COLUMNS-0}" | tr -d '\0'
+    return $_omp_status_cache
 }
 
 if [[ "$TERM" != "linux" ]] && [[ -x "$(command -v ::OMP::)" ]] && ! [[ "$PROMPT_COMMAND" =~ "_omp_hook" ]]; then

--- a/src/shell/scripts/omp.fish
+++ b/src/shell/scripts/omp.fish
@@ -3,12 +3,12 @@ set --export POSH_SHELL_VERSION $FISH_VERSION
 set --global POWERLINE_COMMAND oh-my-posh
 set --global POSH_PID $fish_pid
 set --global CONDA_PROMPT_MODIFIER false
-set --global omp_tooltip_command ''
-set --global omp_current_rprompt ''
-set --global omp_transient false
+set --global _omp_tooltip_command ''
+set --global _omp_current_rprompt ''
+set --global _omp_transient false
 
 # We use this to avoid unnecessary CLI calls for prompt repaint.
-set --global omp_new_prompt true
+set --global _omp_new_prompt true
 
 # template function for context loading
 function set_poshcontext
@@ -24,37 +24,37 @@ function fish_prompt
     # commandline --function repaint does not do this
     # see https://github.com/fish-shell/fish-shell/issues/8418
     printf \e\[0J
-    if test "$omp_transient" = true
-        ::OMP:: print transient --config $POSH_THEME --shell fish --status $omp_status_cache --pipestatus="$omp_pipestatus_cache" --execution-time $omp_duration --stack-count $omp_stack_count --shell-version $FISH_VERSION --no-status=$omp_no_exit_code
+    if test "$_omp_transient" = true
+        ::OMP:: print transient --config $POSH_THEME --shell fish --status $_omp_status_cache --pipestatus="$_omp_pipestatus_cache" --execution-time $_omp_duration --stack-count $_omp_stack_count --shell-version $FISH_VERSION --no-status=$_omp_no_exit_code
         return
     end
-    if test "$omp_new_prompt" = false
-        echo -n "$omp_current_prompt"
+    if test "$_omp_new_prompt" = false
+        echo -n "$_omp_current_prompt"
         return
     end
-    set --global omp_status_cache $omp_status_cache_temp
-    set --global omp_pipestatus_cache $omp_pipestatus_cache_temp
-    set --global omp_stack_count (count $dirstack)
-    set --global omp_duration "$CMD_DURATION$cmd_duration"
-    set --global omp_no_exit_code false
+    set --global _omp_status_cache $omp_status_cache_temp
+    set --global _omp_pipestatus_cache $omp_pipestatus_cache_temp
+    set --global _omp_stack_count (count $dirstack)
+    set --global _omp_duration "$CMD_DURATION$cmd_duration"
+    set --global _omp_no_exit_code false
 
     # check if variable set, < 3.2 case
-    if set --query omp_lastcommand; and test -z "$omp_lastcommand"
-        set omp_duration 0
-        set omp_no_exit_code true
+    if set --query _omp_last_command; and test -z "$_omp_last_command"
+        set _omp_duration 0
+        set _omp_no_exit_code true
     end
 
     # works with fish >=3.2
-    if set --query omp_last_status_generation; and test "$omp_last_status_generation" = "$status_generation"
-        set omp_duration 0
-        set omp_no_exit_code true
-    else if test -z "$omp_last_status_generation"
-        # first execution - $status_generation is 0, $omp_last_status_generation is empty
-        set omp_no_exit_code true
+    if set --query _omp_last_status_generation; and test "$_omp_last_status_generation" = "$status_generation"
+        set _omp_duration 0
+        set _omp_no_exit_code true
+    else if test -z "$_omp_last_status_generation"
+        # first execution - $status_generation is 0, $_omp_last_status_generation is empty
+        set _omp_no_exit_code true
     end
 
     if set --query status_generation
-        set --global omp_last_status_generation $status_generation
+        set --global _omp_last_status_generation $status_generation
     end
 
     set_poshcontext
@@ -70,32 +70,32 @@ function fish_prompt
     ::PROMPT_MARK::
 
     # The prompt is saved for possible reuse, typically a repaint after clearing the screen buffer.
-    set --global omp_current_prompt (::OMP:: print primary --config $POSH_THEME --shell fish --status $omp_status_cache --pipestatus="$omp_pipestatus_cache" --execution-time $omp_duration --stack-count $omp_stack_count --shell-version $FISH_VERSION --cleared=$omp_cleared --no-status=$omp_no_exit_code | string collect)
-    echo -n "$omp_current_prompt"
+    set --global _omp_current_prompt (::OMP:: print primary --config $POSH_THEME --shell fish --status $_omp_status_cache --pipestatus="$_omp_pipestatus_cache" --execution-time $_omp_duration --stack-count $_omp_stack_count --shell-version $FISH_VERSION --cleared=$omp_cleared --no-status=$_omp_no_exit_code | string collect)
+    echo -n "$_omp_current_prompt"
 end
 
 function fish_right_prompt
-    if test "$omp_transient" = true
-        set omp_transient false
+    if test "$_omp_transient" = true
+        set _omp_transient false
         return
     end
     # Repaint an existing right prompt.
-    if test "$omp_new_prompt" = false
-        echo -n "$omp_current_rprompt"
+    if test "$_omp_new_prompt" = false
+        echo -n "$_omp_current_rprompt"
         return
     end
-    set omp_new_prompt false
-    set --global omp_current_rprompt (::OMP:: print right --config $POSH_THEME --shell fish --status $omp_status_cache --pipestatus="$omp_pipestatus_cache" --execution-time $omp_duration --stack-count $omp_stack_count --shell-version $FISH_VERSION --no-status=$omp_no_exit_code | string join '')
-    echo -n "$omp_current_rprompt"
+    set _omp_new_prompt false
+    set --global _omp_current_rprompt (::OMP:: print right --config $POSH_THEME --shell fish --status $_omp_status_cache --pipestatus="$_omp_pipestatus_cache" --execution-time $_omp_duration --stack-count $_omp_stack_count --shell-version $FISH_VERSION --no-status=$_omp_no_exit_code | string join '')
+    echo -n "$_omp_current_rprompt"
 end
 
-function postexec_omp --on-event fish_postexec
+function _omp_postexec --on-event fish_postexec
     # works with fish <3.2
     # pre and postexec not fired for empty command in fish >=3.2
-    set --global omp_lastcommand $argv
+    set --global _omp_last_command $argv
 end
 
-function preexec_omp --on-event fish_preexec
+function _omp_preexec --on-event fish_preexec
     if test "::FTCS_MARKS::" = true
         echo -ne "\e]133;C\a"
     end
@@ -130,16 +130,16 @@ function _omp_space_key_handler
     # Get the first word of command line as tip.
     set --local tooltip_command (commandline --current-buffer | string trim -l | string split --allow-empty -f1 ' ' | string collect)
     # Ignore an empty/repeated tooltip command.
-    if test -z "$tooltip_command" || test "$tooltip_command" = "$omp_tooltip_command"
+    if test -z "$tooltip_command" || test "$tooltip_command" = "$_omp_tooltip_command"
         return
     end
-    set omp_tooltip_command $tooltip_command
-    set --local tooltip_prompt (::OMP:: print tooltip --config $POSH_THEME --shell fish --status $omp_status_cache --pipestatus="$omp_pipestatus_cache" --execution-time $omp_duration --stack-count $omp_stack_count --shell-version $FISH_VERSION --command $omp_tooltip_command --no-status=$omp_no_exit_code | string join '')
+    set _omp_tooltip_command $tooltip_command
+    set --local tooltip_prompt (::OMP:: print tooltip --config $POSH_THEME --shell fish --status $_omp_status_cache --pipestatus="$_omp_pipestatus_cache" --execution-time $_omp_duration --stack-count $_omp_stack_count --shell-version $FISH_VERSION --command $_omp_tooltip_command --no-status=$_omp_no_exit_code | string join '')
     if test -z "$tooltip_prompt"
         return
     end
     # Save the tooltip prompt to avoid unnecessary CLI calls.
-    set omp_current_rprompt $tooltip_prompt
+    set _omp_current_rprompt $tooltip_prompt
     commandline --function repaint
 end
 
@@ -156,10 +156,10 @@ function _omp_enter_key_handler
         return
     end
     if commandline --is-valid || test -z (commandline --current-buffer | string trim -l | string collect)
-        set omp_new_prompt true
-        set omp_tooltip_command ''
+        set _omp_new_prompt true
+        set _omp_tooltip_command ''
         if test "::TRANSIENT::" = true
-            set omp_transient true
+            set _omp_transient true
             commandline --function repaint
         end
     end
@@ -171,10 +171,10 @@ function _omp_ctrl_c_key_handler
         return
     end
     # Render a transient prompt on Ctrl-C with non-empty command line buffer.
-    set omp_new_prompt true
-    set omp_tooltip_command ''
+    set _omp_new_prompt true
+    set _omp_tooltip_command ''
     if test "::TRANSIENT::" = true
-        set omp_transient true
+        set _omp_transient true
         commandline --function repaint
     end
     commandline --function cancel-commandline

--- a/src/shell/scripts/omp.zsh
+++ b/src/shell/scripts/omp.zsh
@@ -76,31 +76,29 @@ autoload -Uz add-zsh-hook
 add-zsh-hook precmd prompt_ohmyposh_precmd
 add-zsh-hook preexec prompt_ohmyposh_preexec
 
-# perform cleanup so a new initialization in current session works
-if [[ "$(bindkey " ")" = *"_posh-tooltip"* ]]; then
-  bindkey " " self-insert
-fi
-if [[ "$(zle -lL zle-line-init)" = *"_posh-zle-line-init"* ]]; then
-  zle -N zle-line-init
-fi
-
-function _posh-tooltip() {
-  # https://github.com/zsh-users/zsh-autosuggestions - clear suggestion to avoid keeping it after the newly inserted space
-  if [[ "$(zle -lL autosuggest-clear)" ]]; then
-    # only if suggestions not disabled (variable not set)
-    if [[ ! -v _ZSH_AUTOSUGGEST_DISABLED ]]; then
-      zle autosuggest-clear
+# Prevent incorrect behaviors when the initialization is executed twice in current session.
+function _omp_cleanup() {
+  local omp_widgets=(
+    self-insert
+    zle-line-init
+  )
+  local widget
+  for widget in "${omp_widgets[@]}"; do
+    if [[ "${widgets[_omp_original::$widget]}" ]]; then
+      # Restore the original widget.
+      zle -A _omp_original::$widget $widget
+    elif [[ "${widgets[$widget]}" = user:_omp_* ]]; then
+      # Delete the OMP-defined widget.
+      zle -D $widget
     fi
-  fi
+  done
+}
+_omp_cleanup
+unset -f _omp_cleanup
 
-  zle .self-insert
-
-  # https://github.com/zsh-users/zsh-autosuggestions - fetch new suggestion after the space
-  if [[ "$(zle -lL autosuggest-fetch)" ]]; then
-    # only if suggestions not disabled (variable not set)
-    if [[ ! -v _ZSH_AUTOSUGGEST_DISABLED ]]; then
-      zle autosuggest-fetch
-    fi
+function _omp_render_tooltip() {
+  if [[ "$KEYS" != ' ' ]]; then
+    return
   fi
 
   # Get the first word of command line as tip.
@@ -121,7 +119,7 @@ function _posh-tooltip() {
   zle .reset-prompt
 }
 
-function _posh-zle-line-init() {
+function _omp_zle-line-init() {
   [[ $CONTEXT == start ]] || return 0
 
   # Start regular line editor.
@@ -149,69 +147,53 @@ function _posh-zle-line-init() {
   return ret
 }
 
-function enable_poshtooltips() {
-  zle -N _posh-tooltip
-  bindkey " " _posh-tooltip
+# Helper function for calling a widget before the specified OMP function.
+function _omp_call_widget() {
+  # The name of the OMP function.
+  local omp_func=$1
+  # The remainder are the widget to call and potential arguments.
+  shift
+
+  zle "$@" && shift 2 && $omp_func "$@"
 }
 
-# Helper function for posh::decorate_widget
-# It calls the posh function right after the original definition of the widget
-# $1 is the posh widget name
-# $2.. are the name of the widget to call + potential args
-posh::call_widget()
-{
-  local posh_widget=$1;shift
-  builtin zle "${@}" &&
-  ${posh_widget}
-}
+# Create a widget with the specified OMP function.
+# An existing widget will be preserved and decorated with the function.
+function _omp_create_widget() {
+  # The name of the widget to create/decorate.
+  local widget=$1
+  # The name of the OMP function.
+  local omp_func=$2
 
-# decorate_widget
-# Allows to preserve any user defined value that may have been defined before posh tries to redefine it.
-# Instead, we keep the previous function and decorate it with the posh additions
-# $1: The name of the widget to decorate
-# $2: The name of the posh function to decorate it with
-function posh::decorate_widget() {
-  orig_widget=${1};shift
-  posh_widget=${1};shift
-  # from this point $@ does not have $1 $2 anymore
+  case ${widgets[$widget]:-''} in
+  # Already decorated: do nothing.
+  user:_omp_decorated_*) ;;
 
-  case ${widgets[$orig_widget]:-""} in
-    # Already decorated: do nothing.
-    user:_posh-decorated-*);;
+  # Non-existent: just create it.
+  '')
+    zle -N $widget $omp_func
+    ;;
 
-    # User defined
-    user:*)
-      zle -N $POSH_PID-$orig_widget ${widgets[$orig_widget]#*:}
-      eval "_posh-decorated-${(q)POSH_PID}-${(q)orig_widget}() { posh::call_widget ${(q)posh_widget} ${(q)POSH_PID}-${(q)orig_widget} -- \"\$@\" }"
-      zle -N $orig_widget _posh-decorated-$POSH_PID-$orig_widget;;
-
-    # Built-in
-    builtin:*)
-      eval "_posh-decorated-${(q)POSH_PID}-${(q)orig_widget}() { posh::call_widget ${(q)posh_widget} .${(q)orig_widget} -- \"\$@\" }"
-      zle -N $orig_widget _posh-decorated-$POSH_PID-$orig_widget;;
-
-    # non-existent
-    *)
-      if [[ $orig_widget == zle-* ]] && (( ! ${+widgets[$orig_widget]} )); then
-        # The widget is a zle one and does not exist, we can safely create it
-        # Otherwise, do nothing
-        eval "_posh-decorated-${(q)POSH_PID}-${(q)orig_widget}() { ${(q)posh_widget} }"
-        zle -N $orig_widget _posh-decorated-$POSH_PID-$orig_widget
-      fi
-      ;;
+  # User-defined or builtin: backup and decorate it.
+  *)
+    # Back up the original widget.
+    zle -A $widget _omp_original::$widget
+    eval "_omp_decorated_${(q)widget}() { _omp_call_widget ${(q)omp_func} _omp_original::${(q)widget} -- \"\$@\" }"
+    zle -N $widget _omp_decorated_$widget
+    ;;
   esac
 }
 
-function enable_poshtransientprompt() {
-  posh::decorate_widget zle-line-init _posh-zle-line-init
-}
+# legacy functions
+function enable_poshtooltips() {}
+function enable_poshtransientprompt() {}
 
 if [[ "::TOOLTIPS::" = "true" ]]; then
-  enable_poshtooltips
+  _omp_create_widget self-insert _omp_render_tooltip
 fi
 
 if [[ "::TRANSIENT::" = "true" ]]; then
-  enable_poshtransientprompt
+  _omp_create_widget zle-line-init _omp_zle-line-init
 fi
 
 if [[ "::UPGRADE::" = "true" ]]; then

--- a/src/shell/scripts/omp.zsh
+++ b/src/shell/scripts/omp.zsh
@@ -9,7 +9,7 @@ export ZLE_RPROMPT_INDENT=0
 # set secondary prompt
 PS2="$(::OMP:: print secondary --config="$POSH_THEME" --shell=zsh)"
 
-function _set_posh_cursor_position() {
+function _omp_set_cursor_position() {
   # not supported in Midnight Commander
   # see https://github.com/JanDeDobbeleer/oh-my-posh/issues/3415
   if [[ "::CURSOR::" != "true" ]] || [[ -v MC_SID ]]; then
@@ -36,45 +36,45 @@ function set_poshcontext() {
   return
 }
 
-function prompt_ohmyposh_preexec() {
+function _omp_preexec() {
   if [[ "::FTCS_MARKS::" = "true" ]]; then
     printf "\033]133;C\007"
   fi
 
-  omp_start_time=$(::OMP:: get millis)
+  _omp_start_time=$(::OMP:: get millis)
 }
 
-function prompt_ohmyposh_precmd() {
-  omp_status_cache=$?
-  omp_pipestatus_cache=(${pipestatus[@]})
-  omp_stack_count=${#dirstack[@]}
-  omp_elapsed=-1
-  omp_no_exit_code="true"
+function _omp_precmd() {
+  _omp_status_cache=$?
+  _omp_pipestatus_cache=(${pipestatus[@]})
+  _omp_stack_count=${#dirstack[@]}
+  _omp_elapsed=-1
+  _omp_no_exit_code="true"
 
-  if [ $omp_start_time ]; then
+  if [ $_omp_start_time ]; then
     local omp_now=$(::OMP:: get millis --shell=zsh)
-    omp_elapsed=$(($omp_now - $omp_start_time))
-    omp_no_exit_code="false"
+    _omp_elapsed=$(($omp_now - $_omp_start_time))
+    _omp_no_exit_code="false"
   fi
 
-  if [[ "${omp_pipestatus_cache[-1]}" != "$omp_status_cache" ]]; then
-    omp_pipestatus_cache=("$omp_status_cache")
+  if [[ "${_omp_pipestatus_cache[-1]}" != "$_omp_status_cache" ]]; then
+    _omp_pipestatus_cache=("$_omp_status_cache")
   fi
 
   count=$((POSH_PROMPT_COUNT + 1))
   export POSH_PROMPT_COUNT=$count
 
   set_poshcontext
-  _set_posh_cursor_position
+  _omp_set_cursor_position
 
-  eval "$(::OMP:: print primary --config="$POSH_THEME" --status="$omp_status_cache" --pipestatus="${omp_pipestatus_cache[*]}" --execution-time="$omp_elapsed" --stack-count="$omp_stack_count" --eval --shell=zsh --shell-version="$ZSH_VERSION" --no-status="$omp_no_exit_code")"
-  unset omp_start_time
+  eval "$(::OMP:: print primary --config="$POSH_THEME" --status="$_omp_status_cache" --pipestatus="${_omp_pipestatus_cache[*]}" --execution-time="$_omp_elapsed" --stack-count="$_omp_stack_count" --eval --shell=zsh --shell-version="$ZSH_VERSION" --no-status="$_omp_no_exit_code")"
+  unset _omp_start_time
 }
 
 # add hook functions
 autoload -Uz add-zsh-hook
-add-zsh-hook precmd prompt_ohmyposh_precmd
-add-zsh-hook preexec prompt_ohmyposh_preexec
+add-zsh-hook precmd _omp_precmd
+add-zsh-hook preexec _omp_preexec
 
 # Prevent incorrect behaviors when the initialization is executed twice in current session.
 function _omp_cleanup() {
@@ -105,12 +105,12 @@ function _omp_render_tooltip() {
   local tooltip_command=${${(MS)BUFFER##[[:graph:]]*}%%[[:space:]]*}
 
   # Ignore an empty/repeated tooltip command.
-  if [[ -z "$tooltip_command" ]] || [[ "$tooltip_command" = "$omp_tooltip_command" ]]; then
+  if [[ -z "$tooltip_command" ]] || [[ "$tooltip_command" = "$_omp_tooltip_command" ]]; then
     return
   fi
 
-  omp_tooltip_command="$tooltip_command"
-  local tooltip=$(::OMP:: print tooltip --config="$POSH_THEME" --status="$omp_status_cache" --pipestatus="${omp_pipestatus_cache[*]}" --execution-time="$omp_elapsed" --stack-count="$omp_stack_count" --command="$tooltip_command" --shell=zsh --shell-version="$ZSH_VERSION" --no-status="$omp_no_exit_code")
+  _omp_tooltip_command="$tooltip_command"
+  local tooltip=$(::OMP:: print tooltip --config="$POSH_THEME" --status="$_omp_status_cache" --pipestatus="${_omp_pipestatus_cache[*]}" --execution-time="$_omp_elapsed" --stack-count="$_omp_stack_count" --command="$tooltip_command" --shell=zsh --shell-version="$ZSH_VERSION" --no-status="$_omp_no_exit_code")
   if [[ -z "$tooltip" ]]; then
     return
   fi
@@ -128,8 +128,8 @@ function _omp_zle-line-init() {
   local -i ret=$?
   (( $+zle_bracketed_paste )) && print -r -n - $zle_bracketed_paste[2]
 
-  omp_tooltip_command=''
-  eval "$(::OMP:: print transient --config="$POSH_THEME" --status="$omp_status_cache" --pipestatus="${omp_pipestatus_cache[*]}" --execution-time="$omp_elapsed" --stack-count="$omp_stack_count" --eval --shell=zsh --shell-version="$ZSH_VERSION" --no-status="$omp_no_exit_code")"
+  _omp_tooltip_command=''
+  eval "$(::OMP:: print transient --config="$POSH_THEME" --status="$_omp_status_cache" --pipestatus="${_omp_pipestatus_cache[*]}" --execution-time="$_omp_elapsed" --stack-count="$_omp_stack_count" --eval --shell=zsh --shell-version="$ZSH_VERSION" --no-status="$_omp_no_exit_code")"
   zle .reset-prompt
 
   # Exit the shell if we receive EOT.

